### PR TITLE
Add GoReleaser task to Catalog

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

This task can be used to builds Go binaries for several platforms,
creates a GitHub release and then pushes a Homebrew formula to a tap
repository using [goreleaser](https://github.com/goreleaser/goreleaser).

closes https://github.com/tektoncd/cli/issues/1170

Signed-off-by: vinamra28 <vinjain@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/goreleaser/0.1/goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - name: package
+      description: base package to build in
+    - name: github-token-secret
+      description: name of the secret holding the github-token
+      default: bot-token-github
+    - name: github-token-secret-key
+      description: name of the secret key holding the github-token
+      default: bot-token
+    - name: flags
+      description: flags to use for the test command
```
Comment (chmouel) [inline task/goreleaser/0.1/goreleaser.yaml:32]: I think this command is wrong (it doesnt aply only for the test command), I believe you can just say 

```
flags to pass to `goreleaser release`
```

**Comment by chmouel** [comment]
/lgtm

Looking Good! 🤙🏽,

**Review by vdemeester — APPROVED** [review]

**Pull request merged**